File: zig/test/standalone/windows_entry_points/dllmain.c
#include <windows.h>
#include <stdio.h>

BOOL WINAPI DllMain(HINSTANCE hinstDLL, DWORD fdwReason, LPVOID lpvReserved) {
    if (fdwReason == DLL_PROCESS_ATTACH) {
        fprintf(stderr, "hello from DllMain");
    }
    return TRUE;
}

File: zig/test/standalone/windows_entry_points/dllmain.zig
const std = @import("std");
const windows = std.os.windows;

const DLL_PROCESS_ATTACH = 1;

pub fn DllMain(
    hinstDLL: windows.HINSTANCE,
    fdwReason: windows.DWORD,
    lpReserved: windows.LPVOID,
) windows.BOOL {
    _ = hinstDLL;
    _ = lpReserved;
    switch (fdwReason) {
        DLL_PROCESS_ATTACH => std.debug.print("hello from DllMain", .{}),
        else => {},
    }
    return .TRUE;
}
